### PR TITLE
Add catalog browsing API

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
This PR adds a service for catalog browsing and tests.

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@okucheruk, PTAL.

File: src/main/java/io/javaclasses/buynbye/ShopHierarchyItem.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+public class ShopHierarchyItem {
+
+    private final String id;
+
+    ShopHierarchyItem(String id) {
+
+        checkNotNull(id);
+        this.id = id;
+    }
```
Comment (okucheruk) [inline src/main/java/io/javaclasses/buynbye/ShopHierarchyItem.java]: Lets improve code readability 
```
    ShopHierarchyItem(String id) {
        this.id = checkNotNull(id);
    }
```

File: src/main/java/io/javaclasses/buynbye/ShoppingItem.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A class which represents a simple shop item.
+ */
+public abstract class ShoppingItem {
+
+    private final String id;
+
+    protected ShoppingItem(String id) {
```
Comment (okucheruk) [inline src/main/java/io/javaclasses/buynbye/ShoppingItem.java]: Add JavaDoc

File: src/main/java/io/javaclasses/buynbye/Product.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return this.price;
+    }
+
+    public String description() {
+        return this.description;
+    }
+
+    public String productName() {
+        return this.productName;
+    }
```
Comment (okucheruk) [inline src/main/java/io/javaclasses/buynbye/Product.java]: Let's have a getters JavaDoc for public API.

**Comment by Oleh-Stefanovych** [comment]
@okucheruk, PTAL.

File: src/main/java/io/javaclasses/buynbye/ShopHierarchyItem.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A class which represents a simple shop item.
+ */
+public class ShopHierarchyItem {
+
+    private final String id;
+
```
Comment (okucheruk) [inline src/main/java/io/javaclasses/buynbye/ShopHierarchyItem.java]: Add JavaDoc

File: src/test/java/io/javaclasses/buynbye/CategoryBrowsingServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    @Test
+    void testFetchProductsInCategoryWithPagination()
+            throws CategoryNotFoundException {
+
+        final int pageSize = 20;
+
+        ChunkedList<Product> chunkedList = categoryBrowsingService
+                .fetchProductsInCategory("category_id_which_exists", 20, pageSize);
```
Comment (okucheruk) [inline src/test/java/io/javaclasses/buynbye/CategoryBrowsingServiceTest.java]: Magic number for pageNumber.

**Pull request merged**